src/librustdoc/clean/mod.rs

```diff
@@ -597,6 +598,7 @@ fn clean_generic_param<'tcx>(
                 ty: Box::new(clean_ty(ty, cx)),
                 default: default
                     .map(|ct| Box::new(ty::Const::from_anon_const(cx.tcx, ct.def_id).to_string())),
+                is_host_effect: cx.tcx.has_attr(param.def_id, sym::rustc_host),
```

Comment:
this works if it comes from the HIR, but that's not always true, there is also a code path where the `is_host_effect` comes from a `ty::GenericParamDef` due to coming from a different crate. I don't have tests for this, but that's the reason why we encode this field in `ty::GenericParamDef` instead of just loading the information lazily.

Reply:
I thought we were filtering out most of them.

@fee1-dead do you remember why we put the `is_host_effect` field on `GenericParamDef`? did I ask for it just to call `has_attr` less?

Reply:
`#[rustc_host]` used to be encoded in an earlier version of #115727 (84a490712aa474a146e619cf15afba22fb652445) but since https://github.com/rust-lang/rust/pull/115727#discussion_r1321386767 it no longer is.

Reply:
> You can call tcx.has_attr on a foreign param, though, and we should be encoding attrs for foreign params too.

We currently don't encode attributes on const generics params, and IIRC I added the field for that purpose